model/user.go

```diff
@@ -724,6 +727,29 @@ func (u *User) GetPreferredTimezone() string {
 	return GetPreferredTimezone(u.Timezone)
 }
 
+func (u *User) ToPatch() *UserPatch {
+	return &UserPatch{
+		Username: &u.Username, Password: &u.Password,
+		Nickname: &u.Nickname, FirstName: &u.FirstName, LastName: &u.LastName,
+		Position: &u.Position, Email: &u.Email,
+		Props: u.Props, NotifyProps: u.NotifyProps,
+		Locale: &u.Locale, Timezone: u.Timezone,
+	}
+}
+
+var patchFieldSelector = map[string]func(*UserPatch) **string{
+	"FirstName": func(p *UserPatch) **string { return &p.FirstName },
+	"LastName":  func(p *UserPatch) **string { return &p.LastName },
+	"Nickname":  func(p *UserPatch) **string { return &p.Nickname },
+	"Email":     func(p *UserPatch) **string { return &p.Email },
+	"Position":  func(p *UserPatch) **string { return &p.Position },
+	"Username":  func(p *UserPatch) **string { return &p.Username },
+}
+
+func (u *UserPatch) SetField(fieldName string, fieldValue string) {
+	*patchFieldSelector[fieldName](u) = &fieldValue
+}
+
```

Comment:
Not sure if I'm the only one but I'd opt to sacrifice lines of code and expand on this in return for readability. It took me a while to understand and although it's compact, I'm not sure how readable it is (and could just be me 😅 ).
Would it be cool to change it to something like the following (and helping us get rid of the global variable too).
```suggestion
func (u *UserPatch) SetField(fieldName string, fieldValue string) {
	switch fieldName {
	case "FirstName":
		u.FirstName = &fieldValue
	case "LastName":
		u.LastName = &fieldValue
	case "Nickname":
		u.Nickname = &fieldValue
	case "Email":
		u.Email = &fieldValue
	case "Position":
		u.Position = &fieldValue
	case "Username":
		u.Username = &fieldValue
	}
}

```
I know this a little bit more drawn out and less compact then what you have but I think it's more readable and easier to catch on to.